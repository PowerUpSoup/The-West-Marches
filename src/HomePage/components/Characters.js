import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import ApiContext from '../../ApiContext';

class Characters extends Component {
    static contextType = ApiContext;

    capitalizeFirstLetter(string) {
        return string.charAt(0).toUpperCase() + string.slice(1);
    }

    render() {
        let sessionStorageUser = JSON.parse(sessionStorage.getItem("user"))

        return (
            <div>
                <Link to="/new-character">Create New Character</Link>
                <ul>Your Characters:
                {this.context.characters.map((character, key) => {
                    if (character.user_id === sessionStorageUser.id) {
                        return (
                            <li key={key}>
                                {this.capitalizeFirstLetter(character.name)}
                            </li>
                        )
                    } else {
                        return null
                    }
                })}
                </ul>
            </div>
        )
    }
}

export default Characters;