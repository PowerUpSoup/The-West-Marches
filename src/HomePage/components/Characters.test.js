  
import React from 'react';
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json'
import Characters from './Characters';

describe(`Characters component`, () => {

    it('renders a Characters', () => {
        const wrapper = shallow(
            <BrowserRouter>
                <Characters />
            </BrowserRouter>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})