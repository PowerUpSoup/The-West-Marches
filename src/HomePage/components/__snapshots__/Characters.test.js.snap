// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders a Characters 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Characters />
</Router>
`;
